### PR TITLE
Added foundations for creating forms

### PR DESCRIPTION
**Pull request by TeoMatosevic**
Set foundations for future creation of forms.

### CODE REVIEW

File: bytepit-ui/src/Models/Form/index.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { FormComponentSpecification } from "./FormComponentSpecification";
+import { FormTypes } from "./FormTypes";
+
+export {
+    type FormComponentSpecification,
+    FormTypes
+}
```
Comment (Maaaarko) [inline bytepit-ui/src/Models/Form/index.ts:7]: please add newline

File: bytepit-ui/src/components/Form/FormComponents/TextInput.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React, { useState } from "react"
+import { FormComponentSpecification } from "../../../Models/Form"
+
+const TextInput = (props: FormComponentSpecification) => {
+    const [value, setValue] = useState(props.value)
+
+    const hnadleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (Maaaarko) [inline bytepit-ui/src/components/Form/FormComponents/TextInput.tsx:7]: typo hnadleChange

**Review by Maaaarko — CHANGES_REQUESTED** [review]

**Review by Maaaarko — APPROVED** [review]
LGTM

**Comment by filipides1** [comment]
Seems good

**Pull request merged**